### ISSUE TITLE
Build dApp in development mode on PRs

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/light-client#1746]
## Description
As part of #1511 , after #1745 the build files are exported as an artifact, but the build itself is done on `--mode ghpages`, which is production-optimized. I'd be nice to have the `build_dapp` step take into account the kind of CI run, and build for `development` in PRs, and `ghpages` (production) only on `master`/`tag` builds (the ones that run after the PR is merged/tagged).
Additionally, it'd be good if the file was compressed on `tar.gz`, as it'd more widely available than `.xz` (e.g. on alpine docker images), and the size difference is not that relevant here.

## Acceptance criteria
- PRs's build of dApp are done in `develop` mode
- artifacts are exported as `.tar.gz` compressed files

## Tasks
- [ ]

**Comment by christianbrb** [comment]
Has to happen on CircleCI
- Check if it is PR which runs the flow -> then build development

**Comment by weilbith** [comment]
@andrevmatos [his draft](https://github.com/raiden-network/light-client/pull/1754) has overruled [mine](https://github.com/raiden-network/light-client/pull/1753). Therefore I switch the assignment of this issue as well.

**Comment by andrevmatos** [comment]
Ok, that was just a draft to be able to test my service, but if it works, good. Thanks
